sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -247,11 +247,35 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 		msg := req.GetProcessBundleSplit()
 
 		log.Debugf(ctx, "PB Split: %v", msg)
+		ref := msg.GetInstructionReference()
+		c.mu.Lock()
+		plan, ok := c.active[ref]
+		c.mu.Unlock()
+		if !ok {
+			return fail(id, "execution plan for %v not found", ref)
+		}
+
+		ds := msg.GetDesiredSplits()["0"]
```

Comment:
It's OK if there isn't any user friendly reason, but is there some documentation we can use for why it's "0" the whole time?

Reply:
hm, basically b/c we can only split at the root node that reads from fnapi (aka "0").
Added comment to state this.